parsers/jscript.c

```diff
@@ -1307,24 +1381,108 @@ static bool parseMethods (tokenInfo *const token, tokenInfo *const class)
 						has_methods = true;
 						addToScope (name, class->string);
 						if (has_child_methods)
-							makeJsTag (name, JSTAG_CLASS, NULL);
+							makeJsTag (name, JSTAG_CLASS, NULL, NULL);
 						else
-							makeJsTag (name, JSTAG_PROPERTY, NULL);
+							makeJsTag (name, JSTAG_PROPERTY, NULL, NULL);
 				}
 			}
 		}
-	} while ( isType(token, TOKEN_COMMA) );
+	} while ( isType(token, TOKEN_COMMA) ||
+	          ( is_es6_class && ! isType(token, TOKEN_EOF) ) );
+
+	JSCRIPT_DEBUG_PRINT("Finished parsing methods");
 
 	findCmdTerm (token, false, false);
 
 cleanUp:
 	deleteToken (name);
+	
+	JSCRIPT_DEBUG_LEAVE();
 
 	return has_methods;
 }
 
+static bool parseES6Class (tokenInfo *const token, const tokenInfo *const parent,
+                           const tokenInfo *targetName)
+{
+	JSCRIPT_DEBUG_ENTER();
+
+	tokenInfo * className = newToken ();
+	vString *inheritance = NULL;
+	bool is_anonymous = true;
+	bool is_terminated = true;
+
+	readToken (className);
+	/* optional name */
+	if (isType (className, TOKEN_IDENTIFIER))
+	{
+		readToken (token);
+		is_anonymous = false;
+	}
+	else
+	{
+		copyToken (token, className, true);
+		/* We create a fake name so we have a scope for the members */
+		if (! targetName)
+		{
+			char buf[32];
+			snprintf (buf, sizeof buf, "AnonymousClass%u", ++AnonymousID);
```

Comment:
While working on the C++ parser I've found that it's useful to have "stable" and "non-colliding" anonymous identifiers.
That is, when parsing the same file twice the generated anonymous identifiers should be the same and when
parsing two different files the generated anonymous identifiers should (ideally) be all different. This is useful for tools that parse the source tree in parts (think of an editor that re-parses the current file while it's being edited).
We worked out a generic tool for this purpose. See usage of anonGenerate() at:
https://github.com/universal-ctags/ctags/blob/77886eb5903f4bcc60a075fbb0961b5210293fb3/parsers/cxx/cxx_token.c#L127